packages/kit/src/runtime/client/router.js

```diff
@@ -16,6 +16,18 @@ function find_anchor(node) {
 	return /** @type {HTMLAnchorElement | SVGAElement} */ (node);
 }
 
+/**
+ * @param {HTMLAnchorElement | SVGAElement } node
+ * @returns {URL}
+ */
+function get_href(node) {
+	const svg = typeof node.href === 'object' && node.href.constructor.name === 'SVGAnimatedString';
+	const href = String(svg ? /** @type {SVGAElement} */ (node).href.baseVal : node.href);
```

Comment:
I wouldn't do any conversion. It looks like it's being done to satisfy TypeScript because `node.href` might not be a string if node is an svg element. However, we already know from the ternary operator that it's not. The issue then is that we're checking if the element is an SVG in a way TypeScript doesn't understand. I'd rewrite this whole function to be:
```
/**
 * @param {HTMLAnchorElement | SVGAElement} node
 * @returns {URL}
 */
 function get_href(node) {
	return node instanceof SVGAElement ? new URL(node.href.baseVal, document.baseURI) : new URL(node.href);
}
```